### PR TITLE
Add docstrings

### PR DESCRIPTION
**Pull request by andrewgiessel**

### CODE REVIEW

File: bolt/utils.py [hunk 1]
```diff
@@ -135,6 +135,18 @@ def slicify(slc, dim):
         raise ValueError("Type for slice %s not recongized" % type(slc))
 
 def istransposeable(new, old):
+    """
+    Check to see if a proposed tuple of axes is a valid permutation
+    of an old set of axes.  Checks length, axis repetion, and bounds.
```
Comment (freeman-lab) [inline bolt/utils.py]: Extra spaces after period

File: bolt/spark/swap.py [hunk 2]
```diff
@@ -83,8 +132,20 @@ def _extract(record):
         return rdd.flatMap(_extract)
 
     def getplan(self, size, dtype):
-        """
-        Identify the plan for chunking along each dimension
+        """Identify the plan for chunking along each value-dimension.  This
```
Comment (freeman-lab) [inline bolt/spark/swap.py]: New line after quotes

**Comment by andrewgiessel** [comment]
First draft of docstrings for `swap.py`, `utils.py`, 'shapes.py`, and `stack.py`. 

Comments and criticism welcome at all levels of detail:  typos, formatting, conceptual mistakes, omissions, clarity, and/or under-documentation.

pinging everyone.  @freeman-lab, @andrewosh @jwittenbach

File: bolt/spark/shapes.py [hunk 3]
```diff
@@ -30,6 +38,14 @@ def shape(self):
         return self._barray.shape[:self._barray.split]
 
     def reshape(self, *shape):
+        """
+        Reshape just the keys of a BoltArraySpark, returning a 
+        new BoltArraySpark.
+
+        Parameters                                                                           
+        ----------                                                                           
+        shape : a tuple of new proposed axes.
```
Comment (freeman-lab) [inline bolt/spark/shapes.py]: Here and elsewhere, the format should be
```
name : type
   description...
```
so here it should be
```
shape : tuple
    The set of new proposed axes.
```
make sense?

Comment (andrewgiessel) [inline bolt/spark/shapes.py]: Yup! Sorry for the inconsistency. Will def shore that up

**Comment by andrewgiessel** [comment]
note that this obviously addresses #54.  Commenting for linkage.

**Comment by freeman-lab** [comment]
Fixes look great, thanks @andrewgiessel !

**Pull request merged**